File: School Projects/2017 - C++ Programming (C++)/Ship/Ship.cpp
#include "Ship.h"

Ship::Ship()
{
	this->name = "";
	this->year = 0;
}

Ship::Ship(string name, int year)
{
	this->name = name;
	this->year = year;
}

string Ship::GetName() const
{
	return name;
}

int Ship::GetYear() const
{
	return year;
}

void Ship::SetName(string name)
{
	this->name = name;
}

void Ship::SetYear(int year)
{
	this->year = year;
}

bool Ship::operator>(Ship other)
{
	return year < other.GetYear();
}

ostream &operator << (ostream &strm, const Ship &obj)
{
	return strm << "Ship name: " << obj.GetName() << endl
				<< "Ship year: " << obj.GetYear() << endl;
}

File: School Projects/2017 - Fundementals Of CSC 2 (Java)/Ship/Ship.java
package ship;
/*
    @author v.shydlonok

    Ship class implements the ShipDetails interface.
    Uses name String to keep track of the ships name.
    Uses year int to keep track of the ships year.
    Defines two constructors for setting up default
        values for name and year variables.
    Defines getters and setters for name and year
        variables.
    Implements DisplayShipDetails() method, prints the
        ships name and year that it is from. 
    Implements isOlder(), returns true if the caller 
        ship is older than the ship passed in through
        the parameter, and false otherwise.
*/
public class Ship implements ShipDetails 
{
    protected String name;
    protected int year;
    
    public Ship()
    {
        name = "";
        year = 0;
    }
    
    public Ship(String name,int year)
    {
        this.name = name;
        this.year = year;
    }

    public String getName() {
        return name;
    }

    public int getYear() {
        return year;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setYear(int year) {
        this.year = year;
    }
    
    @Override
    public void DisplayShipDetails() 
    {
        System.out.println("Ship "+getName()+" is from year "+getYear());
    }

    @Override
    public boolean isOlder(ShipDetails ship) 
    {
        boolean older = false;
        if (this.year < ((Ship)ship).year)
            older = true;
        return older;
    }
}
